subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/PlatformDependencyModifiers.java

```diff
@@ -0,0 +1,150 @@
+     *
+     * @param dependency the dependency
+     * @return the modified dependency
+     */
+    <D extends ModuleDependency> D platform(D dependency);
+
+    /**
+     * Takes a given {@code Provider} to a {@link MinimalExternalModuleDependency} and modifies the dependency to select the Platform variant of the given module.
+     *
+     * @param providerConvertibleToDependency the provider
+     * @return a provider to the modified dependency
+     */
+    default Provider<? extends MinimalExternalModuleDependency> platform(ProviderConvertible<? extends MinimalExternalModuleDependency> providerConvertibleToDependency) {
+        return providerConvertibleToDependency.asProvider().map(this::platform);
+    }
+
+    /**
+     * Takes a given {@code Provider} to a {@link ExternalModuleDependency} and modifies the dependency to select the Platform variant of the given module.
+     *
+     * @param providerToDependency the provider
+     * @return a provider to the modified dependency
+     */
+    default Provider<? extends ExternalModuleDependency> platform(Provider<? extends ExternalModuleDependency> providerToDependency) {
+        return providerToDependency.map(this::platform);
+    }
+
+
+    /**
+     * Creates an {@link ExternalModuleDependency} for the given dependency notation and modifies it to select the Enforced Platform variant of the given module.
+     *
+     * @param dependencyNotation dependency notation
+     * @return the modified dependency
+     * @see DependencyFactory#create(CharSequence) Valid dependency notation for this method
+     */
+    default ExternalModuleDependency enforcedPlatform(CharSequence dependencyNotation) {
+        return enforcedPlatform(getDependencyFactory().create(dependencyNotation));
+    }
+
+    /**
+     * Creates an {@link ExternalModuleDependency} for the given group, name and version and modifies it to select the Enforced Platform variant of the given module.
+     *
+     * @param group the group
+     * @param name the name
+     * @param version the version
+     * @return the modified dependency
+     * @see DependencyFactory#create(String, String, String)
+     */
+    default ExternalModuleDependency enforcedPlatform(@Nullable String group, String name, @Nullable String version) {
+        return enforcedPlatform(getDependencyFactory().create(group, name, version));
+    }
+
+    /**
+     * Takes a given {@link ModuleDependency} and modifies it to select the Enforced Platform variant of the given module.
+     *
+     * @param dependency the dependency
+     * @return the modified dependency
+     */
+    <D extends ModuleDependency> D enforcedPlatform(D dependency);
+
+    /**
+     * Takes a given {@link ExternalDependency} and modifies it to select the Enforced Platform variant of the given module.
+     *
+     * @param dependency the dependency
+     * @return the modified dependency
+     */
+    <D extends ExternalDependency> D enforcedPlatform(D dependency);
```

Comment:
I don't see the point of this. `ExternalDependency extends ModuleDependency`.
```suggestion
```

Reply:
yeah, I think I got carried away trying to avoid instanceof checks